VideoLocker/res/values/attrs.xml

```diff
@@ -6,4 +6,610 @@
         <attr name="itemVerticalPadding" format="dimension" />
         <attr name="iconDefaultSize" format="dimension" />
     </declare-styleable>
+
+    <declare-styleable name="IconView">
+        <attr name="iconColor" format="color" />
+        <attr name="iconName" format="enum">
+            <!-- We can't using IconValue indices as the enum values,
+                 as that would necessitate either an append-only
+                 policy to make them constant, or a recalculation of
+                 all greater indices upon the addition of a new
+                 IconValue in the middle of the list. Instead,
+                 we use the hash code of the name as an established
+                 and non-arbitrary mapping. -->
+            <enum name="fa_adjust" value="-1136693357" />
```

Comment:
@hanningni:

1. No, I didn't check for hash collisions, although they are extremely unlikely in such a small sample space. Ideally, in that case there should be a compile error about duplicate enum values, but I see that Android allows it (maybe for making aliases like `fill_parent` and `match_parent`). In that case we'll need to shift to an alternate mapping. If you want I can test and verify that the current hashes are unique, though I'm pretty sure they are.

2. You can't write Java code in XML :wink:. I _have_ made a dynamic mapping in the IconValue enum, and a method to access it, that's used by `IconDrawable`.